### PR TITLE
Fixes #20527 - Expose more information about Arf Report

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by xprazak2** [comment]
Needs the referenced PR.

File: lib/hammer_cli_foreman_openscap/arf_report.rb [hunk 1]
```diff
@@ -17,9 +25,15 @@ class ListCommand < HammerCLIForeman::ListCommand
...
     class InfoCommand < HammerCLIForemanOpenscap::InfoCommand
       output ListCommand.output_definition do
-        field :host_id, _("Host Id")
-        field :openscap_proxy_id, _("Openscap proxy Id")
-        field :openscap_proxy_name, _("Openscap proxy name")
+        from :host do
+          field :id, _("Host Id")
+        end
+        from :openscap_proxy do
+          field :id, _("Openscap proxy Id")
```
Comment (ares) [inline lib/hammer_cli_foreman_openscap/arf_report.rb]: if this plugin is used against older version of foreman_openscap, we shouldn't print it, @tstrachota correct me if I'm wrong, but we can use `:hide_blank => true` which is what we did for [subnet description in core](https://github.com/theforeman/hammer-cli-foreman/blob/master/lib/hammer_cli_foreman/subnet.rb#L24)

if you agree I think we should add it for other fields and ListCommand too

**Comment by xprazak2** [comment]
I added the `:hide_blank => true` to proxy fields

File: lib/hammer_cli_foreman_openscap/arf_report.rb [hunk 2]
```diff
@@ -6,8 +6,16 @@ class ArfReport < HammerCLIForeman::Command
     class ListCommand < HammerCLIForeman::ListCommand
       output do
         field :id, _("Id")
-        field :host, _("Host")
         field :reported_at, _("Reported at")
+        from :host do
+          field :name, _("Host name")
+        end
+        from :openscap_proxy do
+          field :name, _("Openscap proxy name"), :hide_blank => true
```
Comment (ares) [inline lib/hammer_cli_foreman_openscap/arf_report.rb]: it does not seem to work, I'm getting `Error: undefined method `new' for {:hide_blank=>true}:Hash`, we can probably live with empty fields for now so feel free to remove it (sorry for back and forth). good to merge afterwards

**Comment by xprazak2** [comment]
I reverted back.

**Comment by tstrachota** [comment]
@xprazak2 this should work:
```ruby
field :id, _("Openscap proxy Id"), Fields::Field, :hide_blank => true
```
A field type is required when options are provided.

**Comment by ares** [comment]
For consistency with other merged PRs, I think we can keep it like this.

**Comment by tstrachota** [comment]
Btw if the API provides both names and ids for proxy, policy and host I would consider using `Field::Reference` type that displays only names by default and can expand to name + id on request.

**Comment by xprazak2** [comment]
@tstrachota, is this what you meant by using `Field::Reference`?

**Comment by tstrachota** [comment]
@xprazak2 yes, this is it

**Comment by ares** [comment]
Getting `Error: no implicit conversion of Symbol into Integer` again, after I delete `, Fields::Reference, :hide_blank => true` it works fine. As said in above comment, other associations don't have this so for consistency I think we can avoid it. If someone has older version of foreman_openscap, they only get empty fields.

**Comment by tstrachota** [comment]
So, I'm not sure what's format of the data API returns, but if the references are like `:host => {:id => 1, :name => 'host1'}` the field should be
```ruby
field :host, _("Host"), Fields::Reference, :hide_blank => true
```
The block does nothing in this case and you can remove it. I overlooked it last time, I'm sorry.

@ares using the references will be consistent with the foreman plugin, where we use it for all references. 
BTW could you share hammer debug logs for that error, please?

**Comment by xprazak2** [comment]
@tstrachota, I removed the block and now I am getting:

```
TypeError (no implicit conversion of Symbol into Integer):
    /home/vagrant/projects/hammer-cli-foreman/lib/hammer_cli_foreman/output/formatters.rb:43:in `[]'
    /home/vagrant/projects/hammer-cli-foreman/lib/hammer_cli_foreman/output/formatters.rb:43:in `format'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/output/formatters.rb:62:in `block in format'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/output/formatters.rb:62:in `each'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/output/formatters.rb:62:in `inject'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/output/formatters.rb:62:in `format'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/output/adapter/wrapper_formatter.rb:12:in `format'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/output/adapter/table.rb:78:in `block (2 levels) in format_values'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/output/adapter/table.rb:76:in `each'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/output/adapter/table.rb:76:in `inject'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/output/adapter/table.rb:76:in `block in format_values'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/output/adapter/table.rb:75:in `collect'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/output/adapter/table.rb:75:in `format_values'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/output/adapter/table.rb:26:in `print_collection'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/output/output.rb:27:in `print_collection'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/abstract.rb:157:in `print_collection'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/apipie/command.rb:62:in `print_data'
    /home/vagrant/projects/hammer-cli-foreman/lib/hammer_cli_foreman/commands.rb:275:in `execute'
    /home/vagrant/.rvm/gems/ruby-2.2.4/gems/clamp-1.1.2/lib/clamp/command.rb:63:in `run'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/abstract.rb:24:in `run'
    /home/vagrant/.rvm/gems/ruby-2.2.4/gems/clamp-1.1.2/lib/clamp/subcommand/execution.rb:11:in `execute'
    /home/vagrant/.rvm/gems/ruby-2.2.4/gems/clamp-1.1.2/lib/clamp/command.rb:63:in `run'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/abstract.rb:24:in `run'
    /home/vagrant/.rvm/gems/ruby-2.2.4/gems/clamp-1.1.2/lib/clamp/subcommand/execution.rb:11:in `execute'
    /home/vagrant/.rvm/gems/ruby-2.2.4/gems/clamp-1.1.2/lib/clamp/command.rb:63:in `run'
    /home/vagrant/projects/hammer-cli/lib/hammer_cli/abstract.rb:24:in `run'
    /home/vagrant/.rvm/gems/ruby-2.2.4/gems/clamp-1.1.2/lib/clamp/command.rb:132:in `run'
    /home/vagrant/projects/hammer-cli/bin/hammer:125:in `<top (required)>'
    /home/vagrant/.rvm/gems/ruby-2.2.4/bin/hammer:22:in `load'
    /home/vagrant/.rvm/gems/ruby-2.2.4/bin/hammer:22:in `<main>'
    /home/vagrant/.rvm/gems/ruby-2.2.4/bin/ruby_executable_hooks:15:in `eval'
    /home/vagrant/.rvm/gems/ruby-2.2.4/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by xprazak2** [comment]
Solved by a small [change](https://github.com/theforeman/hammer-cli-foreman/pull/328)

**Comment by xprazak2** [comment]
A change in hammer was merges so this should now work without problems

**Comment by ares** [comment]
@xprazak2 will this hammer change get to 1.15 base? I suppose we'd like to deliver new version of hammer_cli_foreman_openscap with 1.15 too.

**Comment by xprazak2** [comment]
I do not know, maybe @tstrachota can tell us...

**Comment by tstrachota** [comment]
I can release hammer-cli-foreman 0.11.1 with that change.

**Comment by xprazak2** [comment]
No rush, we have [alternate version without references](https://github.com/theforeman/hammer_cli_foreman_openscap/pull/16) ready as well.

I was also thinking that leaving the attribute names for empty attributes in the output makes it more obvious that there is something missing...

**Comment by ares** [comment]
Could you please rebase for 0.2 release?

**Comment by xprazak2** [comment]
I rebased, but I will close this anyway. Note that current version actually does not show the ids in info command output, so less information is displayed compared with what we have in master. And we display taxonomy keys in info even if they are empty, so additional changes are needed there to keep consistent. We already have #16 merged, which addresses this and does not have any major flaws as far as I can see.

@tstrachota, what request do I need to do to expand the reference fields to name + id in info command in case we decide to go with this in the future?

**Comment by tstrachota** [comment]
There's an option `hammer --show-ids ...` that turns the ids on.

**Pull request closed without merging**